compiler/rustc_trait_selection/src/solve/eval_ctxt.rs

```diff
@@ -548,4 +567,56 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
     pub(super) fn universe(&self) -> ty::UniverseIndex {
         self.infcx.universe()
     }
+
+    pub(super) fn create_next_universe(&self) -> ty::UniverseIndex {
+        self.infcx.create_next_universe()
+    }
+
+    pub(super) fn instantiate_canonical_var(
```

Comment:
feels like this method shouldn't exist :thinking: we shouldn't interact with canonical vars while inside of the solver :thinking:

Reply:
Ah yes, this and others you noted below is a leftover for when I tried to pull methods out of canonicalization code. They can be re-inlined since it has access to infcx now.